### DISCUSSION TITLE
CARLA Traffic Light Data Missing

### DISCUSSION THREAD

**Discussion by dwoltgens** [carla-simulator/carla#6957] [Q&A]
I'm using the OpenDRIVE standalone mode of CARLA, and would like to address a specific traffic light in the simulation, however none of the methods to locate a traffic light by it's ID work. Looking at the TrafficLight actors through the GetActors function, they also don't contain any data specific to the traffic light. I've managed to track the issue down to the `FWorldObserver_GetActorState` function in the file `carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla\Sensor\WorldObserver.cpp`. In the function it states that if the traffic light either doesn't have a group or controller assigned, none of the data for the TrafficLight will be retrieved. I'm guessing that the OpenDRIVE standalone mode doesn't assign the traffic lights to a group, which leads to none of the data being retrieved.

Is there a work-around to assign the traffic lights to a group, or some other method to find a traffic light with its OpenDRIVE ID? Or is the only solution to build a map using the OpenDRIVE file and group the traffic lights in the UE4 Editor?

**Comment by Jinviz** [comment]
In my opinion, the only way is to create a map using an open drive file with traffic light information and assign traffic light to the traffic light group within the UE4 editor. That's what I did when I created custom maps.
